packages/react-components/react-storybook-addon-codesandbox/src/sandbox-utils.ts

```diff
@@ -0,0 +1,107 @@
+  const rootElement = document.querySelector(docsSelector);
+
+  if (!rootElement) {
+    throw new Error(`css selector: ${docsSelector}, doesn't exist `);
+  }
+
+  const showCodeButton = rootElement.querySelector('.docblock-code-toggle');
+  const container = showCodeButton?.parentElement;
+
+  if (!container) {
+    throw new Error(`css selector: '.docblock-code-toggle', doesn't exist `);
+  }
+
+  const classList = (showCodeButton.classList.value + ' with-code-sandbox-button').split(' ');
+
+  // this is needed in dev loop as every hot reload will add new buttons
+  if (process.env.NODE_ENV !== 'production') {
+    const ourButtons = container.querySelectorAll(`.with-code-sandbox-button`);
+    ourButtons.forEach(node => node.remove());
+  }
+
+  return {
+    container,
+    cssClasses: classList,
+  };
+}
+
+const addonConfigDefaults = { requiredDependencies: {}, optionalDependencies: {} };
+export type Data = Pick<Required<ParametersConfig>, 'provider' | 'bundler'> & {
+  storyFile: string;
+  // use originalStoryFn because users can override the `storyName` property.
+  // We need the name of the exported function, not the actual story
+  // https://github.com/microsoft/fluentui-storybook-addons/issues/12
+  // originalStoryFn.name someties looks like this: ProgressBarDefault_stories_Default
+  // just get the "Default"
+  // @TODO - im not sure this is still needed, wasn't able to repro. Can we remove it ?
+  storyExportToken: string;
+  dependencies: Record<string, string>;
+  title: string;
+  description: string;
+};
+
+export function prepareData(context: StoryContext): Data | null {
+  if (!context.parameters.exportToSandbox) {
+    throw new Error('exportToSandbox config parameter cannot be empty');
+  }
+
+  const addonConfig: Required<ParametersConfig> = {
+    ...addonConfigDefaults,
+    ...context.parameters.exportToSandbox,
+  };
+  const { provider, bundler } = addonConfig;
+  const storyFile = context.parameters?.fullSource;
+
+  if (!storyFile) {
+    console.error(
+      dedent`Export to CodeSandbox: Couldn't find source for story ${context.story}. Did you install the babel plugin?`,
+    );
+    return null;
+  }
+
+  const { requiredDependencies, optionalDependencies } = addonConfig;
+  const dependencies = getDependencies(storyFile, requiredDependencies, optionalDependencies);
+
+  const title = 'FluentUI React v9';
+  const description = `Story demo: ${context.title} - ${context.name}`;
+
+  // use originalStoryFn because users can override the `storyName` property.
+  // We need the name of the exported function, not the actual story
+  // https://github.com/microsoft/fluentui-storybook-addons/issues/12
+  // originalStoryFn.name someties looks like this: ProgressBarDefault_stories_Default
+  // just get the "Default"
+  // @TODO - im not sure this is still needed, wasn't able to repro. Can we remove it ?
+  const storyExportToken = context.originalStoryFn.name.split('_stories_').slice(-1).pop();
```

Comment:
would love to get more context on this one folks @ling1726 @george-cz

Reply:
we rely on a lot of internal stuff in storybook, so it's very possible that it no longer happens. However, if the eventual plan is to move away from csb, I don't think we should touch much here .🤷

Reply:
> if the eventual plan is to move away from csb, I don't think we should touch much here

hmm AFAIK it has not much to do with  provider used (codesanbox, stackblitz...). as this does some storybook source code gymnastics (which I cannot repro why they are needed).

lets keep it for now - we can always remove later ;)